analyzers/src/SonarAnalyzer.CSharp/Rules/DeclareTypesInNamespaces.cs

```diff
@@ -54,5 +55,15 @@ public sealed class DeclareTypesInNamespaces : DeclareTypesInNamespacesBase<Synt
                 SyntaxKindEx.RecordClassDeclaration,
                 SyntaxKindEx.RecordStructDeclaration,
                 SyntaxKindEx.FileScopedNamespaceDeclaration);
+
+        protected override bool IsException(SyntaxNode declaration) =>
```

Comment:
The parameter name should be `node` or `syntaxNode` since it can be any type of node.